beets/autotag/__init__.py

```diff
@@ -95,7 +95,9 @@ def apply_metadata(album_info, mapping):
         item.title = track_info.title
 
         if config['per_disc_numbering']:
-            item.track = track_info.medium_index or track_info.index
+            item.track = track_info.medium_index
+            if item.track is None:
+                item.track = track_info.index
```

Comment:
I did not inline the conditional because of its size; I'll add that comment there instead.